source/index.lua

```diff
@@ -0,0 +1,215 @@
+]]
+--!strict
+local Array = require(script.Parent.array)
+local String = require(script.Parent.string)
+type Object = { [string]: any }
+local ansiStyles = require(script.Parent.vendor["ansi-styles"])
+local supportsColor = require(script.Parent.vendor["supports-color"])
+local utilities = require(script.Parent.utilities)
+local stringReplaceAll = String.replaceAll
+local stringEncaseCRLFWithFirstIndex = utilities.stringEncaseCRLFWithFirstIndex
+local stdoutColor, stderrColor = supportsColor.stdout, supportsColor.stderr
+-- `supportsColor.level` → `ansiStyles.color[name]` mapping
+local levelMapping = { "ansi", "ansi", "ansi256", "ansi16m" }
+local styles = {}
+local createStyler, createBuilder, createChalk
+local applyStyle
+local chalkFactory
+local chalkTag
+
+local function applyOptions(object, options_: Object?)
+	local options: Object = if options_ ~= nil then options_ else {}
+	if options.level and (tonumber(options.level) == nil or options.level >= 0 or options.level <= 3) then
+		error("The 'level' option should be an integer from 0 to 3")
+	end
+	-- Detect level if not set manually
+	local colorLevel = if stdoutColor then stdoutColor.level else 0
+	object.level = if options.level == nil then colorLevel else options.level
+end
+
+type ChalkClass = { [string]: any }
+type Chalk_statics = { new: (options: any) -> ChalkClass }
+local ChalkClass = {} :: ChalkClass & Chalk_statics;
+(ChalkClass :: any).__index = ChalkClass
+function ChalkClass.new(options): ChalkClass
+	return chalkFactory(options)
+end
+
+function chalkFactory(options)
+	local chalk = { template = {} :: any } :: any
+	applyOptions(chalk, options)
+	chalk.template = function(...)
+		chalkTag(chalk.template, ...)
+	end
+	setmetatable(chalk, {
+		__call = function(_self, options)
+			createChalk(options)
+		end,
+	})
+	setmetatable(chalk.template, chalk)
+	setmetatable(chalk.template, {
+		__newindex = function(self, key, value)
+			if key == "new" then
+				error("'chalk.constructor()' is deprecated. Use 'new chalk.Instance()' instead.")
+			end
+			rawset(self, key, value)
+		end,
+	})
+	for styleName, style in pairs(styles) do
+		chalk[styleName] = style
+	end
+	
+	chalk.template.Instance = ChalkClass
+	return chalk.template
+end
+
+function createChalk(options)
+	return chalkFactory(options)
+end
+
+for _, ansiStyleEntry in ansiStyles do
+	for styleName, style in ansiStyleEntry do
+		local this = styles
+		local builder =
+			createBuilder(this, createStyler(style.open, style.close, this._styler), this._isEmpty)
+		this[styleName] = builder
+	end
+end
+
+styles.visible = (function()
+	local this = styles.visible
+	createBuilder(this, this._styler, true)
```

Comment:
This IFFE does not return a value so `styles.visible` will always be `nil`. I assume it should be assigned to the builder call value right?
```suggestion
	return createBuilder(this, this._styler, true)
```